infrastructure/sentry-subgraph/src/tiny-keys-airdrop.ts

```diff
@@ -1,9 +1,133 @@
-import { AirdropSegmentComplete } from "../generated/TinyKeysAirdrop/TinyKeysAirdrop"
+import { BigInt, log } from "@graphprotocol/graph-ts";
+import { AirdropSegmentComplete, AirdropSegmentStakeComplete } from "../generated/TinyKeysAirdrop/TinyKeysAirdrop"
+import { PoolInfo, SentryWallet, SentryKey, PoolStake } from "../generated/schema"
 
 export function handleAirdropSegmentComplete(event: AirdropSegmentComplete): void {
     // Nothing to handle here
-    if(event){
+    if (event) {
         return;
     }
     return;
-}
\ No newline at end of file
+}
+
+/**
+ * Handles the completion of an airdrop segment staking event. This function updates the relevant 
+ * entities such as PoolInfo, SentryWallet, SentryKey, and PoolStake based on the event details.
+ *
+ * @param {AirdropSegmentStakeComplete} event - The event triggered when a segment of keys is staked 
+ * as part of the tiny keys airdrop. Contains various parameters including the owner address, pool address, and key IDs.
+ *
+ * @returns {void} This function does not return any value.
+ */
+
+export function handleAirdropStakeSegmentComplete(event: AirdropSegmentStakeComplete): void {
+
+    if (event) {
```

Comment:
Can the event be undefined ?

Reply:
I don't believe it can. This came from copying the framework of the function above it in the file.

This has been updated/removed!